google_drive/tests/unit/test_docs.py

```diff
@@ -0,0 +1,240 @@
+        assert_equals(block.alt_text, "This is alt text")
+
+    def test_check_document_url(self):  # pylint: disable=no-self-use
+        """ Test verification of the provided Google Document URL"""
+        block = TestGoogleDocumentBlock.make_document_block()
+
+        data = json.dumps({
+            'url': (
+                "https://docs.google.com/presentation/d/1x2ZuzqHsMoh1epK8VsGAl"
+                "anSo7r9z55ualwQlj-ofBQ/embed?start=true&loop=true&delayms=10000"
+            )
+        })
+        res = block.handle('check_url', make_request(data))
+        # pylint: disable=no-value-for-parameter
+        assert_equals(json.loads(res.body), {'status_code': 200})
+
+        data = json.dumps({
+            'url': 'undefined'
+        })
+        res = block.handle('check_url', make_request(data))
+
+        assert_equals(json.loads(res.body), {'status_code': 400})
+
+        data = json.dumps({
+            'url': (
+                "https://docs.google.com/presentation/d/1x2ZuzqHsMoh1epK8VsdsadfG"
+                "AlanSo7r9z55ualwQlj-ofBQ/embed?start=true&loop=true&delayms=10000"
+            )
+        })
+        res = block.handle('check_url', make_request(data))
+
+        assert_equals(json.loads(res.body), {'status_code': 404})
+
+    def test_document_publish_event(self):  # pylint: disable=no-self-use
+        """ Test event publishing in GoogleDocumentBlock"""
+        block = TestGoogleDocumentBlock.make_document_block()
+
+        body = json.dumps({
+            'url': (
+                'https://docs.google.com/presentation/d/1x2ZuzqHsMoh1epK8VsdsadfGAlanSo7r9z55ualwQlj-ofBQ/embed'
+                '?start=true&loop=true&delayms=10000'
+            ),
+            'displayed_in': 'iframe',
+            'event_type': 'edx.googlecomponent.document.displayed',
+        })
+        res = block.handle('publish_event', make_request(body))
+        # pylint: disable=no-value-for-parameter
+        assert_equals(json.loads(res.body), {'result': 'success'})
+
+        body = json.dumps({
+            'url': (
+                'https://docs.google.com/drawings/d/1LHGzCTLRb--CDvFFjoYp62TiIN5KgsE7QOy9Sift_eg/'
+                'pub?w=882&amp;h=657'),
+            'displayed_in': 'img',
+            'event_type': 'edx.googlecomponent.document.displayed',
+        })
+        res = block.handle('publish_event', make_request(body))
+
+        assert_equals(json.loads(res.body), {'result': 'success'})
+
+        body = json.dumps({
+            'url': (
+                'https://docs.google.com/presentation/d/1x2ZuzqHsMoh1epK8VsdsadfGAlanSo7r9z55ualwQlj-ofBQ/embed'
+                '?start=true&loop=true&delayms=10000'
+            ),
+            'displayed_in': 'iframe',
+        })
+        res = block.handle('publish_event', make_request(body))
+
+        assert_equals(json.loads(res.body), {'result': 'error', 'message': 'Missing event_type in JSON data'})
+
+
+class TestGoogleCalendarBlock(unittest.TestCase):
```

Comment:
Yeah, I would definitely split this test case out into its own test_calendar.py file.  You might also consider having a TestGoogleDriveBlock parent for generalizations -- might make it easier to add support for testing new types of Drive components going forward.